code/_expts/slotnick_etal_2001/de_StimCreate.m

```diff
@@ -19,70 +19,186 @@
 if (~exist('opt','var')),      opt      = {};     end;
 if (~iscell(opt)),             opt      = {opt};  end;
 
-% Create the input images.
-train.nInput = [68 50];
-train.X = zeros(prod(train.nInput), 0);
-train.XLAB = cell(20, 1); % Slotnick has 20 stimuli: 10 far/off, 5 near/off, 5 near/on
-train.TLAB = cell(20, 1);
-%train.XLAB = guru_csprintf('height=%.2f', num2cell(heights);
-counter = 0;
-heights = [];
-if (~strcmp(stimSet, 'blob-dot')) %default set of images
-    error('Invalid stimulus type. Choose one from: {''blob-dot''}');
+if (~strcmp(stimSet, 'blob-dot') && ~strcmp(stimSet, 'paired-squares')) %default set of images
+    error('Invalid stimulus type. Choose one from: {''blob-dot'', ''paired-squares''}');
 end
+
 figure('Position', [0, 0, 1024, 800]);
-% first, 10 "far" images
-for ii=1:5
-    distances = [10, 12];
-    for ij=1:length(distances)
-        img = blob_stimuli(distances(ij), 3, ii);
-        train.X(:, end+1) = reshape(img, prod(train.nInput), 1);
-        counter = counter + 1;
-        train.XLAB{counter} = sprintf('%dpx from %d%c', distances(ij), (ii-1)* 72, char(176));
-        heights = [heights, distances(ij)];
+
+if (strcmp(stimSet, 'blob-dot'))
```

Comment:
Better to use a `switch` statement. And rather than having long blocks of independent code, better to define sub-functions for each stimulus set, and call them.